pkg/controller/rds/dbparametergroup/setup.go

```diff
@@ -0,0 +1,79 @@
+package dbparametergroup
+
+import (
+	"context"
+
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/rds"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/rds/v1alpha1"
+
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+// SetupDBParameterGroup adds a controller that reconciles DBParametergroup.
+func SetupDBParameterGroup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(svcapitypes.DBParameterGroupGroupKind)
+	opts := []option{
+		func(e *external) {
+			e.postCreate = postCreate
+			e.preObserve = preObserve
+			e.preUpdate = preUpdate
+			e.preDelete = preDelete
+			e.postObserve = postObserve
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&svcapitypes.DBParameterGroup{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.DBParameterGroupGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.DescribeDBParameterGroupsInput) error {
+	obj.DBParameterGroupName = awsclients.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.DescribeDBParameterGroupsOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+	cr.SetConditions(xpv1.Available())
+	return obs, err
+}
+
+func postCreate(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.CreateDBParameterGroupOutput, cre managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	meta.SetExternalName(cr, awsclients.StringValue(obj.DBParameterGroup.DBParameterGroupName))
+	return managed.ExternalCreation{ExternalNameAssigned: true}, nil
+}
```

Comment:
We need this logic only if the ID/name is decided by AWS, but in this case we give it to AWS. Similar to `preUpdate`, `preDelete` and `preObserve`, you need to implement `preCreate` as well that gives the external name to the payload object.